### ISSUE TITLE
IsSignatureValid doesn't actually verify the signature

### ISSUE THREAD

**Issue by FlorianHockmann** [secana/PeNet#59]
The check for `PeFile.IsSignatureValid` is implemented in [the private method `AuthenticodeInfo.CheckSignature`](https://github.com/secana/PeNet/blob/0458c44a83e3c39214cbb0e7f2cb30d7d4126ede/src/PeNet/Authenticode/AuthenticodeInfo.cs#L67) which basically does this:
```c#
var hash = ComputeAuthenticodeHash();
return SignedHash.SequenceEqual(hash);
```

This shows that the check only compares whether the hash from the signature matches the computed hash of the PE file, but the signature itself isn't checked at all. It should be verified with the certificate.

**Comment by secana** [comment]
Initial idea for a fix by @FlorianHockmann 
https://github.com/FlorianHockmann/PeNet/commit/e73a1de19d43d06b2aab908d155c3bfdca3da916

**Comment by secana** [comment]
It seems that we are running into a known problem: https://github.com/dotnet/corefx/issues/31073

I've verified that this is the case with the following code:

```csharp
using System;
using System.IO;
using System.Security.Cryptography;
using System.Security.Cryptography.Pkcs;

namespace Framework
{
    class Program
    {
        static void Main(string[] args)
        {
            var rawCerts = File.ReadAllBytes("cert.dump");
            var signedCms = new SignedCms();
            signedCms.Decode(rawCerts);

            try
            {
                signedCms.CheckSignature(true);
                Console.WriteLine("Valid siganture.");
            }
            catch(CryptographicException e)
            {
                Console.Error.WriteLine(e.Message);
            }

            Console.ReadKey();
        }
    }
}
```
On .Net Core: "Invalid Siganture."
On .Net Framework: "Valid Signature."

**Comment by secana** [comment]
Not sure how to handle this. Technically .Net Core is right, since DigiCert doesn't comply to the CMS standard. On the other hand, the "real world" standard would be to accept the signature, like .Net Framework does it.

I've tried to update `System.Security.Cryptography.Pkcs` to a preview version (4.6.0-preview.18571.3), but now the `signedCms.Decode(rawCerts)` fails with an `CryptographicException: ASN1 corrupted data.
` exception.

**Comment by secana** [comment]
I've requested help in the CoreFX issue: dotnet/corefx#31073

**Comment by secana** [comment]
Opened an issue in CoreFX: https://github.com/dotnet/corefx/issues/34202

**Comment by secana** [comment]
Fixed in 0.8.2
